### ISSUE TITLE
[Question] Can date take alphabets and work same.

### ISSUE THREAD

**Issue by gkumar9** [segmentio/metalsmith#298]
I have been using metalsmith-permalinks for custom url of my output but been facing a exquisite problem. 
just as date form this hierarchy when used with metalsmith-permalinks, 
```
---
date: 2017-11-12
---
this results in:

2017/
    11/
        12/
            index.html
```

I want something like below this:
```
---
date: usa-newyork-manhattan
---
this results in:

usa/
    newyork/
        manhattan/
                 index.html
```

**Comment by benpolinsky** [comment]
I don't think so. You might want to open an issue in [metalsmith-permalinks](https://github.com/segmentio/metalsmith-permalinks)

However, another solution would be to use different meta keys for each part of each zone. Something like this: 

```yml
---
zone_country: usa
zone_state: newyork
zone_city: manhattan
---
```

Then:

```js
.use(permalinks({
    pattern: '':zone_country/:zone_state/:zone_city"
}))
```

**Comment by gkumar9** [comment]
@benpolinsky thanks man. But, the things is my requirement is dynamic like I might require 3, or 4 or just 1 values, at times hence need to have something which will work everytime.

**Comment by Ajedi32** [comment]
Yeah, it looks like that repo is unmaintained. I'd fork it, but I don't really have the time right now to maintain another side project.

Does it work if you just use `/` in the name?

```
.use(permalinks({
    pattern: ':subpath'
}))
```

```
---
subpath: usa/newyork/manhattan
---
```

**Comment by benpolinsky** [comment]
@Ajedi32  - interesting suggestion.  Didn't for me, but I just tried it quickly using my current setup which has a bunch of custom plugins. Also on Windows (I tried using a backslashes and double backslashes to counter).

To get what @gkumar9 needs, you'd have to inject a custom function for a given linkSet which would take over the processing? I may be able to fork the plugin and take a look sometime this week.
